src/platform.cpp

```diff
@@ -160,6 +172,32 @@ void platform_bind(Ac action, u32 slot, u32 button, f32 value) {
     global_input.bind(action, slot, button, value);
 }
 
+void platform_audio_callback(void *userdata, u8 *stream, int len) {
+    Audio::AudioStruct *audio_struct_ptr = (Audio::AudioStruct *) userdata;
+    game_lib.audio_callback(audio_struct_ptr, stream, len);
+}
+
+void platform_audio_init() {
+    platform_audio_struct = {};
+
+    SDL_AudioSpec want = {};
+    want.freq = Audio::SAMPLE_RATE;
+    want.format = AUDIO_F32;
+    want.samples = 2048;
+    want.channels = 2;
+    want.callback = platform_audio_callback;
+    want.userdata = (void *) &platform_audio_struct;
+
+    SDL_AudioSpec have;
+    platform_audio_struct.dev = SDL_OpenAudioDevice(nullptr, 0, &want, &have, 0);
+    if (platform_audio_struct.dev <= 0) {
+        UNREACHABLE("Unable to initialize audio (%s)", SDL_GetError());
+    }
+    platform_audio_struct.active = true;
+
```

Comment:
If we are to be "correct" you should check wat is in the "have" struct, so we get what we ask for. We probably do but it's good practice.